src/shared/cache/ClinicalAttributeCache.ts

```diff
@@ -0,0 +1,82 @@
+import client from '../api/cbioportalClientInstance';
```

Comment:
This would be a big change, so don't make it until @alisman comments on this too. 

I think you can simplify this cache a lot. Keep in mind, every request you make on the frontend is already cached until you refresh the page. When the same request is made, the cached version is used. If you're curious how this works, you can look at `apiClientCache` and `config.ts`.

With this in mind, I think what I would do is make a method that takes in a request for a bunch of different clinical data types, splits that request into 1 request per clinical data type, makes those requests and then merges the result. This way each clinical data request is cached on its own by the apiClientCache.

We can talk about how to design this on Monday if Aaron agrees it's a good idea.

Reply:
@onursumer can you weight in?   i can't totally recall how all this works.  if we are allowing sort by these column fields, would we not need to fetch everything up front?

Reply:
As far as I remember, we support delaying data fetch until the user decides to sort by a specific column. If we really want to lazy load clinical attribute columns by page, then we need to have this cache. 

Also we need to make sure that download functionality is working as expected for the clinical data columns. Currently data seems to be empty in the download file for not yet loaded data. I don't remember exactly where/how but we need to update some download data fetcher to make sure that all the data is fetched before starting the data download.

Reply:
Aaron and I chatted and came to the conclusion that @Rajat-Sirohi's current solution is the best way forward. Resolving this issue.